homeassistant/components/whois/config_flow.py

```diff
@@ -0,0 +1,52 @@
+"""Config flow to configure the Whois integration."""
+from __future__ import annotations
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_DOMAIN, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+
+class WhoisFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow for Whois."""
+
+    VERSION = 1
+
+    imported_name: str | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+        if user_input is not None:
+            await self.async_set_unique_id(user_input[CONF_DOMAIN].lower())
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(
+                title=self.imported_name or user_input[CONF_DOMAIN],
+                data={
+                    CONF_DOMAIN: user_input[CONF_DOMAIN].lower(),
+                },
+            )
```

Comment:
I don't agree.

> to verify the user_intput domain format,
> to check if the TLD is supported by python-whois,

We cannot, your suggestion also doesn't do that. The supported domain isn't checkable with this library.

I am replacing the library in a followup pr for that reason.

> and to check if the whois info contains an expiration_date.

That is actually incorrect behavior currently. A domain can be valid without that, additionally a missing state is not a reason to error, it should become an entity with the unknown state instead.

Reply:
> I am replacing the library in a followup pr

It would have been nice if that was in the PR description. Allowing any user_input cannot be a good idea as many users will try e.g. https://www.home-assistant.io when they copy a URL. It's OK if checks are added in another PR.

Reply:
> I am replacing the library in a followup pr

It would have been nice if that was in the PR description. Allowing any user_input cannot be a good idea as many users will try e.g. https://www.home-assistant.io when they copy a URL. It's OK if checks are added in another PR.

Reply:
The PR description stated it was a bare minimum implementation of the config flow

Reply:
I've updated the PR description to make that more clear now.